### PR TITLE
Infer request/response visibility from the method

### PR DESCRIPTION
**Pull request by elb3k**
#### References to other Issues or PRs

https://github.com/grpc-ecosystem/grpc-gateway/issues/2691

#### Brief description of what is fixed or changed
Visibility of message can be inferred from the visibility of method. 
Should solve the problem in https://github.com/grpc-ecosystem/grpc-gateway/issues/2691

### CODE REVIEW

**Review by johanbrandhorst — COMMENTED** [review]
I'm not sure this is right either, what if I am reusing a message between RPCs and only one of the RPCs is hidden? We'd also need to do the inverse, that is, hide an RPC if a message is hidden. I'm still worried about hiding messages that are used in other packages too. I'm not sure there is a satisfying solution to this problem that isn't very complicated.

File: protoc-gen-openapiv2/internal/genopenapi/template.go [hunk 1]
```diff
@@ -1009,10 +1016,34 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
...
+
+			swgRspName, ok := fullyQualifiedNameToOpenAPIName(meth.ResponseType.FQMN(), reg)
+			if !ok && !skipRenderingRef(meth.ResponseType.FQMN()) {
+				glog.Errorf("couldn't resolve OpenAPI name for FQMN '%v'", meth.ResponseType.FQMN())
+				continue
+			}
+
 			if !isVisible(getMethodVisibilityOption(meth), reg) {
+				// Set request/response as invisible if not set
+				if reg.GetMessageVisibility(swgReqName) == descriptor.MessageVisibility_NOT_SET {
```
Comment (elb3k) [inline protoc-gen-openapiv2/internal/genopenapi/template.go:1034]: I one is reusing a message between RPCs and only one of the RPCs is hidden, message will not be hidden. It is handled here.

`RenderServices` looks up all RPCs, and marks message as visible if one or more of RPC uses it.

**Comment by elb3k** [comment]
> I'm not sure this is right either, what if I am reusing a message between RPCs and only one of the RPCs is hidden? We'd also need to do the inverse, that is, hide an RPC if a message is hidden. I'm still worried about hiding messages that are used in other packages too. I'm not sure there is a satisfying solution to this problem that isn't very complicated.

I will test the case with different packages.

**Comment by elb3k** [comment]
> I'm not sure this is right either, what if I am reusing a message between RPCs and only one of the RPCs is hidden? We'd also need to do the inverse, that is, hide an RPC if a message is hidden. I'm still worried about hiding messages that are used in other packages too. I'm not sure there is a satisfying solution to this problem that isn't very complicated.

**Comment by johanbrandhorst** [comment]
@elb3k, I am grateful that you want to contribute to the repository, but I closed the last PR because we need to go back to the drawing board on this design before we can start implementing it. This PR has several issues and we should attempt to sort out everything as part of the design so we don't waste both of ours time reviewing things that are not going to work. Please do not open another PR, lets discuss options in #2691.

**Comment by elb3k** [comment]
> @elb3k, I am grateful that you want to contribute to the repository, but I closed the last PR because we need to go back to the drawing board on this design before we can start implementing it. This PR has several issues and we should attempt to sort out everything as part of the design so we don't waste both of ours time reviewing things that are not going to work. Please do not open another PR, lets discuss options in #2691.

ok

**Pull request closed without merging**